### PR TITLE
Introducing cluster upsert functionality

### PR DESCRIPTION
**Pull request by Mikhail-Ivanov**
Introducing cluster upsert functionality and create/edit methods with DTO parameters
Deprecating old API

### CODE REVIEW

File: src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java [hunk 1]
```diff
@@ -125,4 +153,18 @@ public ClusterInfoDTO getInfo(String clusterId) throws IOException, DatabricksRe
     ClusterEventsDTO clusterEvents = this.mapper.readValue(responseBody, ClusterEventsDTO.class);
     return clusterEvents.getEvents();
   }
+
+  @Override
+  public void upsertCluster(UpsertClusterDTO clusterDTO) throws IOException, DatabricksRestException {
+    String clusterId = clusterDTO.getClusterId();
```
Comment (samshuster) [inline src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java]: Could we modify the upsertClusterDTO to take a clusterName (or perhaps we should support both) and then do a lookup here?
Ideally, the cluster configurations are configured by name and not by id so we can use the same name in multiple environments.

Comment (Mikhail-Ivanov) [reply src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java]: Done, now we do lookup within upsertCluster method

File: src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java [hunk 2]
```diff
@@ -125,4 +153,30 @@ public ClusterInfoDTO getInfo(String clusterId) throws IOException, DatabricksRe
...
     return clusterEvents.getEvents();
   }
+
+  @Override
+  public void upsertCluster(NewClusterDTO clusterDTO) throws IOException, DatabricksRestException {
+    boolean clusterExists = false;
+    String clusterId = "";
+    ClusterInfoDTO[] clusters = list();
+    String clusterName = clusterDTO.getClusterName();
+    for (ClusterInfoDTO cluster : clusters) {
```
Comment (samshuster) [inline src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java]: How about refactor this lookup to its own method and add to interface? its a useful operation.
I don't believe you can have a case with multiple *running* clusters with same name... but that would be good to verify. For jobs for example, there is different logic depending on what people want.

Comment (Mikhail-Ivanov) [reply src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java]: Done, for multiple clusters I've chosen error logging + updating first one

**Pull request merged**